hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieClustering.java

```diff
@@ -0,0 +1,128 @@
+    @Parameter(names = {"--base-path", "-sp"}, description = "Base path for the table", required = true)
+    public String basePath = null;
+    @Parameter(names = {"--table-name", "-tn"}, description = "Table name", required = true)
+    public String tableName = null;
+    @Parameter(names = {"--instant-time", "-it"}, description = "Clustering Instant time", required = true)
+    public String clusteringInstantTime = null;
+    @Parameter(names = {"--parallelism", "-pl"}, description = "Parallelism for hoodie insert", required = true)
+    public int parallelism = 1;
+    @Parameter(names = {"--schema-file", "-sf"}, description = "path for Avro schema file", required = true)
+    public String schemaFile = null;
+    @Parameter(names = {"--spark-master", "-ms"}, description = "Spark master", required = false)
+    public String sparkMaster = null;
+    @Parameter(names = {"--spark-memory", "-sm"}, description = "spark memory to use", required = true)
+    public String sparkMemory = null;
+    @Parameter(names = {"--retry", "-rt"}, description = "number of retries", required = false)
+    public int retry = 0;
+    @Parameter(names = {"--help", "-h"}, help = true)
+    public Boolean help = false;
+
+    @Parameter(names = {"--props"}, description = "path to properties file on localfs or dfs, with configurations for "
+        + "hoodie client for clustering")
+    public String propsFilePath = null;
+
+    @Parameter(names = {"--hoodie-conf"}, description = "Any configuration that can be set in the properties file "
+        + "(using the CLI parameter \"--props\") can also be passed command line using this parameter. This can be repeated",
+            splitter = IdentitySplitter.class)
+    public List<String> configs = new ArrayList<>();
+  }
+
+  public static void main(String[] args) {
+    final Config cfg = new Config();
+    JCommander cmd = new JCommander(cfg, null, args);
+    if (cfg.help || args.length == 0) {
+      cmd.usage();
+      System.exit(1);
+    }
+    final JavaSparkContext jsc = UtilHelpers.buildSparkContext("clustering-" + cfg.tableName, cfg.sparkMaster, cfg.sparkMemory);
+    HoodieClustering clustering = new HoodieClustering(jsc, cfg);
+    clustering.cluster(cfg.retry);
+  }
+
+  public int cluster(int retry) {
+    this.fs = FSUtils.getFs(cfg.basePath, jsc.hadoopConfiguration());
+    int ret = -1;
+    try {
+      do {
+        ret = scheduleAndCluster(jsc);
+      } while (ret != 0 && retry-- > 0);
+    } catch (Throwable t) {
+      LOG.error(t);
+    }
+    return ret;
+  }
+
+  private int scheduleAndCluster(JavaSparkContext jsc) throws Exception {
+    String schemaStr = new Schema.Parser().parse(fs.open(new Path(cfg.schemaFile))).toString();
+    SparkRDDWriteClient client =
+        UtilHelpers.createHoodieClient(jsc, cfg.basePath, schemaStr, cfg.parallelism, Option.empty(), props);
+    client.scheduleClusteringAtInstant(cfg.clusteringInstantTime, Option.empty());
```

Comment:
schedule returns a boolean. We can only run clustering if return value is true. You may want to add that check 
boolean isScheduled =     client.scheduleClusteringAtInstant(cfg.clusteringInstantTime, Option.empty());
if (isScheduled) {
// run clustering
} else {
// log that there is no enough data to run clustering
}

Reply:
have separate  schedule and cluster